test/unit/mdc-textfield/mdc-text-field.test.js

```diff
@@ -352,6 +352,47 @@ test('#adapter.deregisterTextFieldInteractionHandler removes an event handler fo
   td.verify(handler(td.matchers.anything()));
 });
 
+test('#adapter.registerValidationAttributeChangeHandler creates a working mutation observer', (done) => {
+  const {root, component} = setupTest();
+  const handler = td.func('ValidationAttributeChangeHandler');
+  td.when(handler(td.matchers.anything(), td.matchers.anything())).thenDo(() => {
+    assert.isNotOk(root.classList.contains('mdc-text-field--invalid'));
+    done();
+  });
+
+  component.foundation_.adapter_.registerValidationAttributeChangeHandler(handler);
+  root.classList.add('mdc-text-field--invalid');
+  root.querySelector('.mdc-text-field__input').required = true;
+});
+
+test('#adapter.registerValidationAttributeChangeHandler adds an observer to the component', () => {
+  const {component} = setupTest();
+  const handler = td.func('ValidationAttributeChangeHandler');
+
+  component.foundation_.adapter_.registerValidationAttributeChangeHandler(handler);
+  assert.equal(component.observers_.length, 2);
+});
+
+test('#adapter.deregisterValidationAttributeChangeHandler disconnects observers from component', () => {
```

Comment:
I don't think we need both this and the next test? One of them should be enough.